tripal_chado/src/ChadoBuddy/ChadoBuddyPluginBase.php

```diff
@@ -43,4 +44,75 @@ public function description(): string {
     return (string) $this->pluginDefinition['description'];
   }
 
+  /**
+   * Used to validate input arrays to various buddy functions,
+   * or to generate a valid array subset.
+   * The output has the column aliases de-aliased to the
+   * actual chado column names, e.g. ['db_name' => 'db.name']
+   * indicates that the alias 'db_name' corresponds to the
+   * column 'name' in the 'db' table.
+   *
+   * @param array $user_values
+   *   An associative array to be validated.
+   * @param array $valid_values
+   *   An associative array listing all valid keys, and the
+   *   values are un-aliased database table alias and table
+   *   column as described above.
+   * @param bool $filter
```

Comment:
This function is a bit confusing as it appears to do two things. I think it would be more clear if you change this $filter parameter to a string where it's value is the table name. If filter is provided then only valid columns from that table are returned and the key is "unaliases" but if that parameter is NULL then the user_values are returned as is but an exception is thrown for invalid columns.

Reply:
This is now split into two separate functions, so it should be much clearer.